### PR TITLE
feat: Add vector similarity search and traverse similar_to relationships

### PR DESCRIPTION
**Pull request by ferdeleong**
- Add vector embeddings to `amenity` nodes as properties
- Create vector index in amenities using the vector embedding property
- Implement vector similarity search 
- Retrieve up to 2 nodes connected by `SIMILAR_TO` relationship

### CODE REVIEW

File: retrieval_service/datastore/providers/neo4j_graph.py [hunk 1]
```diff
@@ -169,20 +196,62 @@ async def search_airports(
...
+                MATCH (amenity: Amenity {id: $id})
+                RETURN amenity.id AS id,
+                    amenity.name AS name,
+                    amenity.description AS description,
+                    amenity.location AS location,
+                    amenity.terminal AS terminal,
+                    amenity.category AS category,
+                    amenity.hour AS hour
+                """,
+                id=id,
```
Comment (ferdeleong) [inline retrieval_service/datastore/providers/neo4j_graph.py:209]: specify return fields so embedding wont be included in it

**Review by duwenxin99 — APPROVED** [review]

File: retrieval_service/datastore/providers/neo4j_graph.py [hunk 2]
```diff
@@ -69,12 +69,16 @@ async def initialize_data(
         async def delete_graph(tx):
             await tx.run("MATCH (n) DETACH DELETE n")
 
+        async def delete_vector_index(tx):
+            # Delete amenity vector index
+            await tx.run("DROP INDEX embedding_amenity IF EXISTS")
```
Comment (averikitsch) [inline retrieval_service/datastore/providers/neo4j_graph.py:74]: nit: can this variable be `amenity_embedding`?

File: retrieval_service/datastore/providers/neo4j_graph.py (hunk 1 above)
Comment (averikitsch) [reply retrieval_service/datastore/providers/neo4j_graph.py:209]: Can you add this as a comment?

File: retrieval_service/datastore/providers/neo4j_graph.py [hunk 3]
```diff
@@ -169,20 +196,62 @@ async def search_airports(
...
+                """,
+                query_embedding=query_embedding,
+                top_k=top_k,
+            )
+
+            amenities = []
+            async for record in result:
+                amenities.append(dict(record))
+
+            print(amenities)
```
Comment (averikitsch) [inline retrieval_service/datastore/providers/neo4j_graph.py:253]: nit: remove print

File: retrieval_service/datastore/providers/neo4j_graph.py [hunk 4]
```diff
@@ -169,20 +196,62 @@ async def search_airports(
...
+                        targetAmenity.hour AS target_hour
+                LIMIT 2
+                """,
+                query_embedding=query_embedding,
+                top_k=top_k,
+            )
+
+            amenities = []
+            async for record in result:
+                amenities.append(dict(record))
```
Comment (averikitsch) [inline retrieval_service/datastore/providers/neo4j_graph.py:251]: Can we just call result.data()?

Comment (averikitsch) [reply retrieval_service/datastore/providers/neo4j_graph.py:251]: Or [record.data() for record in results]?

File: retrieval_service/datastore/providers/neo4j_graph_test.py [hunk 5]
```diff
@@ -174,3 +175,60 @@ async def test_total_similar_to_relationships_count(ds: neo4j_graph.Client):
...
+
+amenities_search_test_data = [
+    pytest.param(
+        # "Where can I look for luxury goods?"
+        amenities_query_embedding2,
+        None,  # similarity threshold value
+        2,  # top_k value
+        [
+            {
+                "source_name": "Gucci Duty Free",
```
Comment (averikitsch) [inline retrieval_service/datastore/providers/neo4j_graph_test.py:188]: Does this test ensure that the vector search is working ie would this test work just using the graph relationships?

**Review by averikitsch — APPROVED** [review]

File: retrieval_service/datastore/providers/neo4j_graph_test.py (hunk 5 above)
Comment (ferdeleong) [reply retrieval_service/datastore/providers/neo4j_graph_test.py:188]: To test the graph relationships we need to use vector search to find the initial node and then traverse it.

**Pull request merged**